Based on the FTD predictions that we applied to the GINCO, I'll analyse how the FTD labels are connected with GINCO labels.

In [1]:
import pandas as pd
import numpy as np

In [3]:
# Import the GINCO dataset
ginco = pd.read_csv("final_data/GINCO-MT-GINCO-keeptext-split-file-with-all-information.csv", sep="\t", index_col=0)
ginco.head(3)

id                                                url  crawled   hard  \
0  3949  http://www.pomurje.si/aktualno/sport/zimska-li...     2014  False   
1  3726  http://www.ss-sezana.si/sss/index.php?option=c...     2014  False   
2  5621  http://www.kamnik-starejsi.si/novice/144-sodel...     2014  False   

           primary_level_1          primary_level_2          primary_level_3  \
0           News/Reporting           News/Reporting           News/Reporting   
1  Information/Explanation  Information/Explanation  Information/Explanation   
2    Promotion of Services    Promotion of Services                Promotion   

       secondary_level_1      secondary_level_2      secondary_level_3  ...  \
0                    NaN                    NaN                    NaN  ...   
1                    NaN                    NaN                    NaN  ...   
2  Opinion/Argumentation  Opinion/Argumentation  Opinion/Argumentation  ...   

          tertiary_level_3  split                  domain  \
0                      NaN   test          www.pomurje.si   
1                      NaN  train        www.ss-sezana.si   
2  Information/Explanation  train  www.kamnik-starejsi.si   

                   GINCORE                                       Slovene_text  \
0                     News  Šport <p/> Zimska liga malega nogometa sobota,...   
1  Information/Explanation  JEDILNIK <p/> Iskalnik <p/> Poglavitni cilj pr...   
2                Promotion  Projekt INNOVAge in zavod Oreli <p/> Zavod Ore...   

                                             MT_text text_length  \
0  Sport <p/> Winter Little League Football Satur...          93   
1  JEDILNIK <p/> Search <p/> The main objective o...          76   
2  Project INNOVAge and the Oreli Institute <p/> ...         232   

      FTD_pred_on_SL     FTD_pred_on_MT split-without-rare-categories  
0          A8 (news)          A8 (news)                          test  
1  A16 (information)  A16 (information)                          test  
2    A12 (promotion)    A12 (promotion)                         train  

[3 rows x 22 columns]

In [6]:
# See label distribution in GINCO
ginco.primary_level_1.value_counts()

Information/Explanation       130
News/Reporting                115
Promotion of a Product        115
Opinion/Argumentation         114
List of Summaries/Excerpts    106
Opinionated News               89
Forum                          52
Instruction                    38
Other                          34
Invitation                     32
Promotion of Services          32
Promotion                      30
Legal/Regulation               17
Announcement                   17
Review                         17
Correspondence                 16
Call                           11
Research Article                9
Interview                       8
Recipe                          6
Prose                           6
Lyrical                         4
FAQ                             3
Script/Drama                    1
Name: primary_level_1, dtype: int64

In [9]:
# Analyse how GINCO primary labels are connected with FTD labels based on the prediction
pd.crosstab(ginco["primary_level_1"], ginco["FTD_pred_on_MT"], normalize="index")


FTD_pred_on_MT              A1 (argumentative)  A11 (personal)  \
primary_level_1                                                  
Announcement                          0.000000        0.000000   
Call                                  0.272727        0.000000   
Correspondence                        0.312500        0.187500   
FAQ                                   0.000000        0.000000   
Forum                                 0.384615        0.403846   
Information/Explanation               0.046154        0.007692   
Instruction                           0.000000        0.026316   
Interview                             0.250000        0.375000   
Invitation                            0.000000        0.000000   
Legal/Regulation                      0.000000        0.000000   
List of Summaries/Excerpts            0.141509        0.047170   
Lyrical                               0.000000        0.750000   
News/Reporting                        0.069565        0.000000   
Opinion/Argumentation                 0.456140        0.184211   
Opinionated News                      0.202247        0.000000   
Other                                 0.117647        0.088235   
Promotion                             0.000000        0.000000   
Promotion of Services                 0.031250        0.000000   
Promotion of a Product                0.000000        0.000000   
Prose                                 0.000000        0.166667   
Recipe                                0.000000        0.166667   
Research Article                      0.111111        0.000000   
Review                                0.058824        0.117647   
Script/Drama                          0.000000        0.000000   

FTD_pred_on_MT              A12 (promotion)  A14 (academic)  \
primary_level_1                                               
Announcement                       0.411765        0.000000   
Call                               0.272727        0.000000   
Correspondence                     0.125000        0.000000   
FAQ                                0.333333        0.000000   
Forum                              0.057692        0.000000   
Information/Explanation            0.253846        0.000000   
Instruction                        0.131579        0.000000   
Interview                          0.250000        0.000000   
Invitation                         0.812500        0.000000   
Legal/Regulation                   0.000000        0.000000   
List of Summaries/Excerpts         0.264151        0.009434   
Lyrical                            0.000000        0.000000   
News/Reporting                     0.034783        0.008696   
Opinion/Argumentation              0.070175        0.000000   
Opinionated News                   0.044944        0.000000   
Other                              0.294118        0.000000   
Promotion                          0.800000        0.000000   
Promotion of Services              0.906250        0.000000   
Promotion of a Product             0.826087        0.000000   
Prose                              0.000000        0.000000   
Recipe                             0.000000        0.000000   
Research Article                   0.000000        0.777778   
Review                             0.176471        0.000000   
Script/Drama                       0.000000        0.000000   

FTD_pred_on_MT              A16 (information)  A17 (review)  A4 (fiction)  \
primary_level_1                                                             
Announcement                         0.117647      0.000000      0.000000   
Call                                 0.000000      0.000000      0.000000   
Correspondence                       0.000000      0.062500      0.000000   
FAQ                                  0.000000      0.000000      0.000000   
Forum                                0.038462      0.019231      0.000000   
Information/Explanation              0.569231      0.015385      0.007692   
Instruction                 

In [ ]:
pd.crosstab(ginco["primary_level_1"], ginco["FTD_pred_on_MT"], normalize="index").to_dict("index")

What we can see based on the FTD predictions on the GINCO labels, is (first, the information for prediction on Slovene text is given, followed by the information what is different on predictions on the MT text):

1. Some labels are very well connected (similar):
    - Instruction (38): very well identified as 'A7 (instruction)': 0.79; on MT text even better identified: 'A7 (instruction)': 0.84
    - Invitation (32): very well identified as  'A12 (promotion)': 0.875; on MT slightly worse identified as 'A12 (promotion)': 0.81, 20% identified as News: 'A8 (news)': 0.1875,
    - Legal/Regulation (17): very well identified as 'A9 (legal)': 1.0, on MT the same
    - Promotion of Services (32): very well predicted as  'A12 (promotion)': 0.94; a bit worse predicted on MT: 'A12 (promotion)': 0.91
    - Promotion of a Product (115): very well predicted as 'A12 (promotion)': 0.85; a bit worse predicted on MT: 'A12 (promotion)': 0.83
    - Promotion (30): very well predicted as  'A12 (promotion)': 0.8; the same on MT
    - Prose (6): very well predicted as 'A4 (fiction)': 1.0; on MT a bit worse predicted: 'A4 (fiction)': 0.83, 'A11 (personal)': 0.17
    - Recipe (6): very well predicted as 'A7 (instruction)': 0.83; the same on MT
    - Research Article (9): well predicted as  'A14 (academic)': 0.67, otherwise as 'A16 (information)': 0.22; on MT even better prediction: 'A14 (academic)': 0.77, otherwise 'A16 (information)': 0.11
    - Review (17): predicted as 'A17 (review)': 0.65, otherwise 'A12 (promotion)': 0.29; on MT similar prediction of review, but less promotion: 'A12 (promotion)': 0.17, some 'A11 (personal)': 0.12 instead

2. Some seem to be less connected:
    - Information/Explanation (130 instances): relatively well identified as 'A16 (information)': 0.6, a third identified as  'A12 (promotion)': 0.32; on MT text similarly identified as 'A16 (information)': 0.57, 25% identified as 'A12 (promotion)'.
    - News/Reporting (115): half identified as 'A8 (news)': 0.55, otherwise also 'A1 (argumentative)': 0.10,  'A12 (promotion)': 0.13,   'A16 (information)': 0.10 and other categories; on MT text much better identification of News: 'A8 (news)': 0.79
    - Opinion/Argumentation (114): less than half identified as 'A1 (argumentative): 0.39, otherwise 'A11 (personal)': 0.14,  'A12 (promotion)': 0.12,  'A16 (information)': 0.17,  'A17 (review)': 0.13 and others; on MT text similar: mostly 'A1 (argumentative)': 0.46, 'A11 (personal)': 0.18
    - Opinionated News (89): half identified as 'A8 (news)': 0.528, otherwise mostly 'A1 (argumentative)': 0.13 or 'A12 (promotion)': 0.12 or  'A17 (review)': 0.17; on MT even more identified as News: 'A8 (news)': 0.63, 20% as  'A1 (argumentative)'


3. Some GINCO labels were predicted by a variety of FTD labels - are not well connected:
    - Announcement (17 instances): not closely connected to just one FTD label - instances were predicted as 'A12 (promotion)': 0.47 (of all Announcement instances), 'A16 (information)': 0.18, 'A7 (instruction)': 0.06,   'A8 (news)': 0.29; on MT text, the main label are 'A8 (news)': 0.41 and 'A12 (promotion)': 0.41
    - Call (11 instances): mostly connected to 'A12 (promotion)': 0.73; on MT text, it is divided between 'A1 (argumentative)', 'A12 (promotion)' and 'A9 (legal)' (0.27)
    - Correspondence (16 instances): not well predicted with FTDs - mostly split between 'A1 (argumentative)': 0.375, 'A7 (instruction)': 0.125, and 'A8 (news)': 0.0625 - on MT similar, but there is more 'A7 (instruction)': 0.25 and also 13% 'A12 (promotion)'
    - FAQ (3 instances): mostly predicted as 'A12 (promotion)': 0.67 - to note: there are very little FAQ instances in the corpus. On MT text, it is mostly predicted as 'A7 (instruction)': 0.66 and 33 % as 'A12 (promotion)'.
    - Forum (52 instances): not well connected with FTD categories - identified as 'A1 (argumentative)': 0.38,  'A11 (personal)': 0.19,'A17 (review)': 0.15, also some   'A12 (promotion)', 'A14 (academic)',  'A16 (information)', 'A17 (review)',  'A7 (instruction)'. On MT text it is mostly 'A11 (personal)': 0.40 and 'A1 (argumentative)': 0.38.
    - Interview (8): not well connected to just one category: 'A1 (argumentative)': 0.125, 'A11 (personal)': 0.125,   'A12 (promotion)': 0.375,  'A16 (information)': 0.125,  'A17 (review)': 0.25. On MT text, predictions are scattered between 'A1 (argumentative)': 0.25, 'A11 (personal)': 0.375, 'A12 (promotion)': 0.25, and 'A17 (review)': 0.125.
    - List of Summaries/Excerpts (106): most vague category, not connected to FTD categories well - identified as all of the FTD categories; on MT the same.
    - Lyrical (4): half identified as 'A11 (personal)': 0.5, half as  'A4 (fiction)': 0.5; on MT the same combination of labels, but more identified as 'A11 (personal)': 0.75
    - Other (34): as I assumed, the label was predicted with various FTD labels, mostly 'A12 (promotion)': 0.44, 'A16 (information)': 0.15 or  'A17 (review)': 0.21; on MT even more scattered, less promotion ('A12 (promotion)': 0.29)
    - Script/Drama (1): not well predicted, but there is only one instance. Predicted as  'A16 (information)': 1.0; the same on MT

Comparison between the prediction on Slovene and MT text shows that mostly there is not a big difference between prediction on Slovene or English text. On some labels, the predictions are worse on MT (Promotion labels, Prose), on some it is better (Instruction, Research Article, Information/Explanation). On MT, there is a much better identification of News and Opinionated News. In other cases, there is no difference or just a slight difference.